src/main/java/com/limechain/client/FullNode.java

```diff
@@ -39,17 +43,20 @@ public class FullNode implements HostNode {
     public void start() {
         //Initialize state
 
-        // TODO: Is there a better way to do this?
+        // TODO: Is there a better way to decide whether we've got any database written?
         var db = AppBean.getBean(KVRepository.class); //presume this works
         // if: database exists and has some persisted storage
         if (db != null && db.find(new BlockStateHelper().headerHashKey(BigInteger.ZERO)).isPresent()) {
             // do nothing?
         } else {
-            List<InsertTrieNode> trie = new InsertTrieBuilder()
-                    .initializeTrieStructure(loadGenesisStorage())
-                    .build();
+            Genesis genesisStorage = loadGenesisStorage();
 
-            insertStorage(db, trie, InsertTrieBuilder.STATE_VERSION); //Todo: calculate state version
+            // TODO: Next up, manage to fetch the runtime version
+            StateVersion stateVersion = StateVersion.V0;
+
+            TrieStructure<NodeData> trie = TrieStructureFactory.buildFromKVPs(genesisStorage.getTop(), stateVersion);
+            List<InsertTrieNode> dbSerializedTrieNodes = new InsertTrieBuilder(trie).build();
+            insertStorage(db, dbSerializedTrieNodes, InsertTrieBuilder.STATE_VERSION); //Todo: calculate state version
```

Comment:
The state version passed here is different than the one set above.

Reply:
True, fixed this.